sdk/src/main/java/io/opentelemetry/sdk/trace/data/LinkData.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.trace.config.TraceConfig;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.SpanContext;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * An immutable implementation of {@link Link}.
+ *
+ * @since 0.1.0
+ */
+@Immutable
+@AutoValue
+public abstract class LinkData implements Link {
```

Comment:
This is inconsistent, SpanData represents the interface to access recorded properties to the Span, but LinkData represents a concrete implementation of the Link interface.

Reply:
Totally agree. Do you have ideas on better naming? I didn't like "Link" which is what it was before. And, we already have another class called "TimedEvent" for the other case.